src/shippingState.js

```diff
@@ -0,0 +1,271 @@
+    const providerBoxWithSpecimenData = addSpecimenDataToDetailBox(providerBoxesObj, finalizedSpecimenList);
+    const detailedProviderBoxes = addBoxDataToDetailBox(providerBoxWithSpecimenData, boxesByProviderList);
+    const detailedLocationBoxes = filterDetailBoxesByLocation(detailedProviderBoxes);
+
+    appState.setState({
+        allBoxesList: allBoxesList,
+        availableCollectionsObj: availableCollectionsObj,
+        availableLocations: availableLocations,
+        boxesByLocationList: boxesByLocationList,
+        detailedLocationBoxes: detailedLocationBoxes,
+        detailedProviderBoxes: detailedProviderBoxes,
+        finalizedSpecimenList: finalizedSpecimenList,
+        userName: userName,
+    });
+}
+
+export const updateShippingStateSelectedLocation = (selectedLocation) => {
+    appState.setState({
+        selectedLocation: selectedLocation,
+    });
+
+    const detailedProviderBoxes = appState.getState().detailedProviderBoxes;
+    const detailedLocationBoxes = filterDetailBoxesByLocation(detailedProviderBoxes);
+
+    appState.setState({
+        detailedLocationBoxes: detailedLocationBoxes,
+    });
+}
+
+/**
+ * Update the state for allBoxesList and availableCollectionsObj.
+ * Remove the bags from the box.
+ * @param {string} boxId - the box with the bag to be removed.
+ * @param {array} bagsToMove - the bags to be removed from the box. 
+ */
+export const updateShippingStateRemoveBagFromBox = (boxId, bagsToMove) => {
+    addBagToAvailableCollections(boxId, bagsToMove);
+    removeBagFromBox(boxId, bagsToMove);
+}
+
+/**
+ * Update the state for allBoxesList and availableCollectionsObj.
+ * Add the bag to the box.
+ * @param {string} boxId - the box with the bag to be added
+ * @param {array} bagToMove - the bag to be added to the box
+ */
+export const updateShippingStateAddBagToBox = (boxId, bagId, boxToUpdate) => {
+    addBagToBox(boxId, boxToUpdate);
+    removeBagFromAvailableCollections(bagId);
+}
+
+/**
+ * The bag is being removed from the box. Put it back in the availableCollectionsObj
+ * Build the bag and specimen data from detailedProviderBoxes data: availableCollectionsObj[bagId]: ['tube1Id', 'tube2Id', 'tube3Id']
+ * Isolate the last four digits of tubeId to match availableCollectionsObj format.
+ * Move object into availableCollectionsObj
+ * @param {*} boxId - the box with the bag being removed 
+ * @param {*} bagsToMove - the bags being removed from the box
+ */
+const addBagToAvailableCollections = (boxId, bagsToMove) => {
+    const availableCollectionsObj = appState.getState().availableCollectionsObj;
+    const detailedProviderBoxes = appState.getState().detailedProviderBoxes;
+
+    for (const bagId of bagsToMove) {
+        const collectionToMove = detailedProviderBoxes[boxId][bagId].arrElements;
+        const tubeIdArray = collectionToMove.map(tubeId => tubeId.slice(-4));
```

Comment:
I like defining `const tubeIdArray` here because of the clarity/readability - mostly because we were recently alerted that some of these data structures may change in the upcoming 6-12 months. I definitely see your point about it being an unnecessary extra step. Let's leave this open in case anyone else has feedback.